server/http/helloapp/helloapp.go

```diff
@@ -21,26 +30,142 @@ type helloapp struct {
 func Init(router *mux.Router, apps *apps.Service) {
 	a := helloapp{
 		mm:           apps.Mattermost,
-		configurator: apps.Config,
+		configurator: apps.Configurator,
 	}
 
 	subrouter := router.PathPrefix(constants.HelloAppPath).Subrouter()
 	subrouter.HandleFunc("/mattermost-app.json", a.handleManifest).Methods("GET")
+
+	subrouter.HandleFunc("/wish/install", a.handleInstall).Methods("POST")
 }
 
 func (h *helloapp) handleManifest(w http.ResponseWriter, req *http.Request) {
 	conf := h.configurator.GetConfig()
 
+	rootURL := conf.PluginURL + constants.HelloAppPath
+
 	httputils.WriteJSON(w,
 		apps.Manifest{
 			AppID:       "hello",
 			DisplayName: "Hallo სამყარო",
 			Description: "Hallo სამყარო test app",
-			RootURL:     conf.PluginURL + constants.HelloAppPath,
+			RootURL:     rootURL,
 			RequestedPermissions: []apps.PermissionType{
 				apps.PermissionUserJoinedChannelNotification,
 				apps.PermissionActAsUser,
 				apps.PermissionActAsBot,
 			},
+			Install: &apps.Wish{
+				URL: rootURL + "/wish/install",
+			},
+		})
+}
+
+func (h *helloapp) handleInstall(w http.ResponseWriter, req *http.Request) {
+	authValue := req.Header.Get(apps.AuthHeader)
+	if !strings.HasPrefix(authValue, "Bearer ") {
+		httputils.WriteBadRequestError(w, errors.Errorf("missing %s: Bearer header", apps.AuthHeader))
+		return
+	}
+
+	jwtoken := strings.TrimPrefix(authValue, "Bearer ")
+	claims := apps.JWTClaims{}
+	_, err := jwt.ParseWithClaims(jwtoken, &claims, func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte(AppSecret), nil
+	})
+	if err != nil {
+		httputils.WriteBadRequestError(w, err)
+		return
+	}
```

Comment:
We will probably need this in more places. Maybe extract this to a middleware for the HandleFuncs?

Can be done in a different PR, and due to this being a POC we may just copy paste where needed.